### DISCUSSION TITLE
💡 [IDEA] - Unified, Unambiguous Transport Independent TPN Broadcast Protocol

### DISCUSSION THREAD

**Discussion by ty-everett** [bsv-blockchain/BRCs#87] [Ideas]
### Summary

We need to align the ecosystem onto a single, unified, transport-independent protocol for transaction broadcast to the Transaction Processing Network, including proof acquisition, double-spend handling, and clear procedures [such as retrying or dropping] for all parties in case of all possible errors.

Currently, the ARC API is the closest we have. But having **ANY** ambiguity whatsoever when engaging with the Transaction Processing Network is fundamentally problematic.

### Example

Sending transactions to the Transaction Processing Network is currently an ambiguous process, with both undefined failure modes and undefined partial and full success cases. This alignment would mitigate these issues.

### Relevance to BSV

- [X] This proposal is relevant to the Bitcoin SV network

**Comment by ty-everett** [comment]
Unless we align on making this new protocol THE exclusive way to interact with the TPN, we should also account for other, out-of-band interactions.

Broadly, I can conceive of three types of interactions:
* Transaction Ingest Messages (inwards, from the Transacting Parties to the TPN)
* Transaction Responses [success and error] (outwards, from the Transaction Processors to the Transacting Parties)
* Unsolicited Callbacks (again, outwards), for Inclusion Notifications and Double Spend Alerts

**Comment by rob-tkn** [comment]
A minimal "model client" example would be helpful.
